polkadot/node/network/collator-protocol/src/validator_side/mod.rs

```diff
@@ -1086,30 +1076,48 @@ where
 	let (collator_id, para_id) = peer_data
 		.insert_advertisement(
 			relay_parent,
-			relay_parent_mode,
 			candidate_hash,
 			&state.implicit_view,
 			&state.active_leaves,
+			&per_relay_parent,
 		)
 		.map_err(AdvertisementError::Invalid)?;
 
-	if per_relay_parent.collations.is_seconded_limit_reached(relay_parent_mode) {
+	let claims_for_para = per_relay_parent.collations.claims_for_para(&para_id);
+	let seconded_and_pending_at_ancestors = seconded_and_pending_for_para_in_view(
+		&state.implicit_view,
+		&state.per_relay_parent,
+		&relay_parent,
+		&para_id,
+	);
+
+	gum::trace!(
+		target: LOG_TARGET,
+		?relay_parent,
+		?para_id,
+		claims_for_para,
+		seconded_and_pending_at_ancestors,
+		"Checking if seconded limit is reached"
+	);
+
+	// Checks if another collation can be accepted. The number of collations that can be seconded
+	// per parachain is limited by the entries in claim queue for the `ParaId` in question.
+	if seconded_and_pending_at_ancestors >= claims_for_para {
```

Comment:
We return SecondedLimitReached Error. There is still a possibility that the pending collation will not be seconded for some reason. What happens then to the 1 extra slot that could be used for seconding?